app/validators/declaration_date_within_milestone_validator.rb

```diff
@@ -0,0 +1,21 @@
+class DeclarationDateWithinMilestoneValidator < ActiveModel::Validator
+  def validate(record)
+    declaration_within_milestone(record)
+  end
+
+private
+
+  def declaration_within_milestone(record)
```

Comment:
I would leave this inlined; its not that complex and we aren't going to reuse it anywhere else

Reply:
Hi @ethax-ross We have the same validator being used in [declaration model](https://github.com/DFE-Digital/register-early-career-teachers-public/blob/main/app/models/declaration.rb#L58).. so that's why I extracted it..

Reply:
Ah so it is 👍